Deno test shim: Deno.TestContext.name is `undefined`

The following test passes with `deno test` but fails with `deno run -A build-node-package.ts` because `t.name` is `undefined`:

```ts
import { assertEquals } from "std/testing/asserts.ts";

Deno.test("example", (t) => {
  assertEquals(t.name, "example");
});
```

<details>
<summary>example build logs</summary>

```log
[dnt] Transforming...
[dnt] Running npm install...

added 11 packages, and audited 12 packages in 2s

2 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
[dnt] Building project...
[dnt] Type checking...
[dnt] Emitting declaration files...
[dnt] Emitting ESM package...
[dnt] Emitting script package...
[dnt] Running tests...

> test
> node test_runner.js

Running tests in ./script/mod.test.js...

test example ... fail

FAILURES

example
  AssertionError: Values are not equal:
  
  
      [Diff] Actual / Expected
  
  
  -   undefined
  +   'example'
  
      at assertEquals (/path/to/dnt-playground/node/script/deps/deno.land/std@0.154.0/testing/asserts.js:178:11)
      at Object.fn (/path/to/dnt-playground/node/script/mod.test.js:29:35)
      at runTestDefinitions (/path/to/dnt-playground/node/test_runner.js:54:30)
      at main (/path/to/dnt-playground/node/test_runner.js:31:15)
      at Object.<anonymous> (/path/to/dnt-playground/node/test_runner.js:200:1)
      at Module._compile (node:internal/modules/cjs/loader:1105:14)
      at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
      at Module.load (node:internal/modules/cjs/loader:981:32)
      at Function.Module._load (node:internal/modules/cjs/loader:822:12)
      at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)error: Uncaught (in promise) Error: npm run test failed with exit code 1
        throw new Error(
              ^
    at runCommand (https://deno.land/x/dnt@0.30.0/lib/utils.ts:56:15)
    at async build (https://deno.land/x/dnt@0.30.0/mod.ts:372:5)
    at async file:///path/to/dnt-playground/build-node-package.ts:5:1

```

</details>

--- a/lib/test_runner/test_runner.ts
+++ b/lib/test_runner/test_runner.ts
@@ -89,7 +89,10 @@ export async function runTestDefinitions(
     options.process.exit(1);
   }
 
-  function getTestContext(definition: TestDefinition, parent: TestContext | undefined): TestContext {
+  function getTestContext(
+    definition: TestDefinition,
+    parent: TestContext | undefined,
+  ): TestContext {
     return {
       name: definition.name,
       parent,

--- a/lib/test_runner/test_runner.ts
+++ b/lib/test_runner/test_runner.ts
@@ -21,13 +21,13 @@ export interface RunTestDefinitionsOptions {
 }
 
 export interface TestDefinition {
-  name: string | undefined;
+  name: string;
   fn: (context: TestContext) => Promise<void> | void;
   ignore?: boolean;
 }
 
 export interface TestContext {
-  name: string | undefined;
+  name: string;
   parent: TestContext | undefined;
   origin: string;
   err: any;
@@ -52,7 +52,7 @@ export async function runTestDefinitions(
       options.process.stdout.write(` ${options.chalk.gray("ignored")}\n`);
       continue;
     }
-    const context = getTestContext(undefined);
+    const context = getTestContext(definition, undefined);
     let pass = false;
     try {
       await definition.fn(context);
@@ -89,9 +89,9 @@ export async function runTestDefinitions(
     options.process.exit(1);
   }
 
-  function getTestContext(parent: TestContext | undefined): TestContext {
+  function getTestContext(definition: TestDefinition, parent: TestContext | undefined): TestContext {
     return {
-      name: undefined,
+      name: definition.name,
       parent,
       origin: options.origin,
       /** @type {any} */
@@ -132,9 +132,7 @@ export async function runTestDefinitions(
       async step(nameOrTestDefinition, fn) {
         const definition = getDefinition();
 
-        const context = getTestContext(this);
-        context.status = "pending";
-        context.name = definition.name;
+        const context = getTestContext(definition, this);
         context.status = "pending";
         this.children.push(context);
 

--- a/lib/test_runner/test_runner.ts
+++ b/lib/test_runner/test_runner.ts
@@ -108,7 +108,7 @@ export async function runTestDefinitions(
       },
       getOutput() {
         let output = "";
-        if (this.name) {
+        if (this.parent) {
           output += "test " + this.name + " ...";
         }
         if (this.children.length > 0) {
@@ -118,16 +118,16 @@ export async function runTestDefinitions(
         } else if (!this.err) {
           output += " ";
         }
-        if (this.name && this.err) {
+        if (this.parent && this.err) {
           output += "\n";
         }
         if (this.err) {
           output += indentText((this.err.stack ?? this.err).toString(), 1);
-          if (this.name) {
+          if (this.parent) {
             output += "\n";
           }
         }
-        if (this.name) {
+        if (this.parent) {
           output += getStatusText(this.status);
         }
         return output;

--- a/tests/test_project/mod.test.ts
+++ b/tests/test_project/mod.test.ts
@@ -8,6 +8,10 @@ Deno.test("should add in test project", () => {
 });
 
 Deno.test("should get properties on test context", async (t) => {
+  if (t.name !== "should get properties on test context") {
+    console.error("Name", t.name);
+    throw new Error("Test definition name was unexpected.");
+  }
   const url = import.meta.url;
   if (t.origin !== url) {
     console.log(`Context origin: ${t.origin}`);
@@ -18,6 +22,10 @@ Deno.test("should get properties on test context", async (t) => {
     throw new Error("Parent should have been undefined.");
   }
   await t.step("inner", (tInner) => {
+    if (tInner.name !== "inner") {
+      console.error("Name", tInner.name);
+      throw new Error("Test step definition name was unexpected.");
+    }
     if (tInner.parent !== t) {
       throw new Error("The parent was not correct.");
     }

